toolchain/check/import_ref.cpp

```diff
@@ -131,30 +131,34 @@ auto VerifySameCanonicalImportIRInst(Context& context, SemIR::InstId prev_id,
 //    - If `retry` is set, we process it again, because it didn't complete last
 //      time, even though we have a constant value already.
 // 2. Resolve the instruction: (TryResolveInst/TryResolveTypedInst)
+//    - For a symbolic constant within a generic, find the generic itself. If it
+//      needs to be imported, return Retry() to import the generic before we
+//      import the constant.
 //    - For instructions that can be forward declared, if we don't already have
 //      a constant value from a previous attempt at resolution, start by making
 //      a forward declared constant value to address circular references.
 //    - Gather all input constants.
 //      - Gathering constants directly adds unresolved values to work_stack_.
 //    - If any need to be resolved (HasNewWork), return Retry(): this
-//      instruction needs two calls to complete.
+//      instruction needs another call to complete.
 //      - If the constant value is already known because we have made a forward
 //        declaration, pass it to Retry(). It will be passed to future attempts
 //        to resolve this instruction so the earlier work can be found, and will
 //        be made available for other instructions to use.
-//      - The second attempt to resolve this instruction must produce the same
-//        constant, because the value may have already been used by resolved
-//        instructions.
+//      - The subsequent attempt to resolve this instruction must produce the
+//        same constant, because the value may have already been used by
+//        resolved instructions.
 //    - Build any necessary IR structures, and return the output constant.
 // 3. If resolve didn't return Retry(), pop the work. Otherwise, it needs to
 //    remain, and may no longer be at the top of the stack; set `retry` on it so
 //    we'll make sure to run it again later.
 //
-// TryResolveInst/TryResolveTypedInst can complete in one call for a given
-// instruction, but should always complete within two calls. However, due to the
-// chance of a second call, it's important to reserve all expensive logic until
-// it's been established that input constants are available; this in particular
-// includes GetTypeIdForTypeConstant calls which do a hash table lookup.
+// TryResolveInst can complete in one call for a given instruction, but should
+// always complete within three calls, and TryResolveTypedInst should always
+// complete within two calls. However, due to the chance of a second call, it's
```

Comment:
```suggestion
// complete within two calls. However, due to the chance of a later call, it's
```

Reply:
Clarified in the changes for the other comment.